src/test/groovy/org/prebid/server/functional/tests/module/requestcorrenction/ResponseCorrectionSpec.groovy

```diff
@@ -0,0 +1,520 @@
+
+        and: "Save account with enabled response correction module"
+        accountDao.save(accountConfigWithResponseCorrectionModule(bidRequest))
+
+        when: "PBS processes auction request"
+        def response = pbsServiceWithResponseCorrectionModule.sendAuctionRequest(bidRequest)
+
+        then: "PBs should emit log"
+        def logsByTime = pbsServiceWithResponseCorrectionModule.getLogsByTime(start)
+        def responseCorrection = getLogsByText(logsByTime, bidId)
+        assert responseCorrection.size() == 1
+        assert responseCorrection.any {
+            it.contains("Bid $bidId of bidder generic has an JSON ADM, that appears to be native" as String)
+        }
+
+        and: "Response should contain seatBid type and mediaType"
+        assert response.seatbid.size() == 1
+        assert response.seatbid[0].bid[0].ext.prebid.type == VIDEO
+
+        and: "Response shouldn't contain errors"
+        assert !response.ext.errors
+
+        and: "Response shouldn't contain warnings"
+        assert !response.ext.warnings
+    }
+
+    def "PBs shouldn't modify response meta.mediaType to video and emit logs when requested impression with video and adm obj with asset"() {
+        given: "Start up time"
+        def start = Instant.now()
+
+        and: "Default bid request with APP and audio imp"
+        def bidRequest = getDefaultBidRequest(APP).tap {
+            imp[0] = Imp.getDefaultImpression(VIDEO)
+        }
+
+        and: "Set bidder response"
+        def bidResponse = BidResponse.getDefaultBidResponse(bidRequest)
+        def bidId = bidResponse.seatbid[0].bid[0].id
+        bidder.setResponse(bidRequest.id, bidResponse)
+
+        and: "Save account with enabled response correction module"
+        accountDao.save(accountConfigWithResponseCorrectionModule(bidRequest))
+
+        when: "PBS processes auction request"
+        def response = pbsServiceWithResponseCorrectionModule.sendAuctionRequest(bidRequest)
+
+        then: "PBs should emit log"
+        def logsByTime = pbsServiceWithResponseCorrectionModule.getLogsByTime(start)
+        def responseCorrection = getLogsByText(logsByTime, bidId)
+        assert responseCorrection.size() == 1
+        assert responseCorrection.any {
+            it.contains("Bid $bidId of bidder generic has an JSON ADM, that appears to be native" as String)
+        }
+
+        and: "Response should contain seatBid type and mediaType"
+        assert response.seatbid.size() == 1
+        assert response.seatbid[0].bid[0].ext.prebid.type == VIDEO
+        assert !response?.seatbid[0]?.bid[0]?.ext?.prebid?.meta?.mediaType
+
+        and: "Response shouldn't contain errors"
+        assert !response.ext.errors
+
+        and: "Response shouldn't contain warnings"
+        assert !response.ext.warnings
+    }
+
+    def "PBs should modify response meta.mediaType to video and type to banner and emit logs when requested impression with native and adm value is not asset"() {
```

Comment:
Maybe: `PBS should modify meta.mediaType and type for original response and also emit logs when response contains native meta.mediaType and adm without asset`